scripts/checkpoint_mapped_reads.py

```diff
@@ -22,11 +22,16 @@ def determine_if_all_segments_aligned(bamstats, reference) -> bool:
 
     bamstats_segments = set()
     with open(bamstats, "r") as f:
-        # skip header line
-        next(f)
-        for line in f:
-            bamstats_segments.add(line.split()[0])
-        f.close()
+        # try to skip header line
+        try:
+            next(f)
+        except StopIteration as err:
+            print("No coverage statistics found in BAMStats log.")
+        else:
+            for line in f:
+                bamstats_segments.add(line.split()[0])
+        finally:
+            f.close()
 
```

Comment:
Something I just learned from @tsibley : You don't actually need to write `f.close()` when you have a `with ... as f:` statement at the top; `with` as a context manager automatically closes the declared variable `f` for you, even on error!

Reply:
Oh! Good to know!